### PR TITLE
Sepia extract rebalance [TM Cand/Ready]

### PR DESCRIPTION
**Pull request by 4dplanner**
## About The Pull Request

reworks sepia extract and some crossbreeds for new effects.

todo: 

- [x] mechs are vulnerable to sepia

- [x] robots are vulnerable to sepia 

- [x] antimagic only works against magic timestop

## Why It's Good For The Game

Point by point:

Core change: sepia slimes are very overpowered currently, being a long stun that ignores stunimmunity, blockchance, doesn't require aiming, and covers a 5x5 area. I think having a delay on them makes them an interesting area denial tool/guaranteed kill if you have a gun.

Recurring sepia: This was insanely powerful. You could run around spamming timefields like there was no tomorrow. However, I didn't want to just make it useless - I think the recalling effect will make sure it fills the intended role of a convenient and sustainable extract. In fact this crossbreed was so overpowered it overshadowed all of the other sepia crossbreeds, meaning they barely got used.

Chilling sepia: As above, but it'll be interesting to see how the teamwork dynamic works out. I think it'll open up some interesting strategies, while being bad for a solo murderboner player.

Burning sepia: The timefreeze camera was as mentioned before overshadowed by the recurring sepia, but is still very powerful. Combined with a laser you can guarantee a kill with a single undodgeable instant travel click at long range, which is crazy.

## Changelog
:cl:
add: sepia slime extract has a delay before activating
balance: the user of a sepia slime extract is no longer immune to the time field
balance: chilling sepia is now the support extract - it always freezes the user, but not other marked people
balance: give one to your golem!
add: burning sepia spawns the rewind camera. Take a selfie with someone and give it to them to make sure they remember the moment forever!
balance: you don't actually need to give them the photo, they'll remember anyway. Keep it as a reminder.
add: rewinding resets your location to the turf you were on after 10 seconds, and heals simple animals and objects.
remove: timefreeze camera is admin spawn only.
add: recurring sepia recalls to the hand of the last person to touch it after activating. Reusable timestop grenades!
Tweak: some special cameras don't prompt for customisation
fix: timefreeze camera actually makes a photo 
fix: timestop stops pathing 
fix: timestop stops mechs
fix: adds a check to make sure simple_animals don't get their AI toggled while sentient
fix: Adds the timestop overlay to frozen projectiles
fix: Timestopped things have INFINITY move_resist as opposed to being anchored
fix: Timestop will now unfreeze things that somehow leave it
fix: mobs in the middle of a walk_to will have their walk stopped by timestop
fix: mobs that are stunned will be stopped mid walk as well
fix: Slimes respect mobility_flags & MOBILITY_MOVE
fix: Slimes no longer automatically regain MOBILITY_MOVE whenever not cold
fix: pulling respects changes in move_force
fix: swapping places respects move_force if the participant is not willing
/:cl:

**Cobby's Note:** I misclicked okay don't worry :)

### CODE REVIEW

**Comment by Repukan** [comment]
You know what this PR needs?

Markass Brawnlee

**Comment by 4dplanner** [comment]
that's hot

**Comment by imsxz** [comment]
how does the rewind camera work? seems really neat from a quick glance at the code but im a brainlet and having trouble grasping it entirely.

**Comment by 4dplanner** [comment]
> how does the rewind camera work? seems really neat from a quick glance at the code but im a brainlet and having trouble grasping it entirely.

You click the guy, he goes around as normal, and ten seconds later he jumps back to the turf he was on when it started.

**Comment by 4dplanner** [comment]
Bonus: if they are a simplemob/an object, they heal up as well.

**Comment by zxaber** [comment]
>mechs are vulnerable to sepia

Farewell to my dabbing on admemes that try to trap me with a timestop, only for me to stomp away freely.

**Comment by ExcessiveUseOfCobblestone** [comment]
Antimagic not working on science timestops will be okay, I'd rather balance with that then "JUST GET A MELON IF YOU HATE XENOBIO" arguments.

**Comment by 4dplanner** [comment]
Looking at the code, they should actually already be immune. Bug maybe?

**Comment by 4dplanner** [comment]
In fact testing seems to suggest that antimagic doesn't work on slime timestop at all, and I'm not really sure why people keep suggesting it as a counter

**Comment by ExcessiveUseOfCobblestone** [comment]
no one (including me) reads code lol

**Comment by knacker48** [comment]
People keep talking like xenobiologist is the most overpowered job yet I swear they always die first since they're so Isolated which makes them an easy kill. Also all of the good stuff usually comes after the threat has passed or everyone's dead and the shuttle left

**Comment by Nickkoranda** [comment]
I haven't really ever seen anybody go out of their way to kill xenobio specifically unless they have a huge hate boner for it (lexia black). Generally, the isolation is what makes it useful since people would rather roam around the central hallway for victims than to break into departments.

**Comment by ExcessiveUseOfCobblestone** [comment]
yeah xenobio is pretty underdeveloped with admin revives, instant death bolts, ability to create an army of golems or simplemobs each with their own unique abilities, ability to start fires, ability to bomb, ability to convert people into borgs, and whatever the latest and greatest wacky crossbreeding is.

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 1]
```diff
@@ -559,9 +559,17 @@
...
+	sleep(50)
+	var/obj/item/slime_extract/sepia/extract = holder.my_atom
 	var/turf/T = get_turf(holder.my_atom)
-	var/list/M = list(get_mob_by_key(holder.my_atom.fingerprintslast))
-	new /obj/effect/timestop(T, null, null, M)
+	new /obj/effect/timestop(T, null, null, null)
+	if(istype(extract))
+		if(extract.Uses > 0)
+			var/mob/lastheld = get_mob_by_key(holder.my_atom.fingerprintslast)
+			if(lastheld)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]:
```suggestion
			if(lastheld && !lastheld.equip_to_slot_if_possible(extract, SLOT_HANDS, disable_warning = TRUE))
```

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/recipes/slime_extracts.dm]: this suggestion is bait since you need to fix 570/1

**Comment by ExcessiveUseOfCobblestone** [comment]
looks ok but cringing at the fact there's like 5 lists that may or may not be used everytime timestop is made

**Comment by 4dplanner** [comment]
Should I compress them down to two lists?

**Comment by 4dplanner** [comment]
Actually probably don't need the throwing list either, will jumble it all in

**Comment by ExcessiveUseOfCobblestone** [comment]
awesome just ping me when it's done and ill merge

**Comment by optimumtact** [comment]
it's more correct to say the rewards are developed, but the actual gameplay of the job is extremely lacking, because slimes are super easy to care for.

**Comment by Shadowflame909** [comment]
@ExcessiveUseOfCobblestone None of those can beat 1 memer boy hunting their target in a place where no one will bother them before an hour in.

And once an hour in hits, all of those can't beat a gygax or a durand.

**Comment by Angustmeta** [comment]
Fixes #37249 , an old issue I found while browsing.

**Comment by knacker48** [comment]
@Angustmeta That's how I see it too. There's a lot of randomness when it comes to how the slimes split. For instance if you are trying to rush rainbow slimes you just have to hope that you get lucky and get a orange and that orange splits into a red. I've had a couple of rounds where I couldn't get a red until the 40 minute mark, at which point I basically gave up on it because the shuttle was called

**Comment by optimumtact** [comment]
that only illustrates the point though.

The only complexities are attention and RNG, there's limited actual deep engagment required.

**Comment by ExcessiveUseOfCobblestone** [comment]
> @ExcessiveUseOfCobblestone None of those can beat 1 memer boy hunting their target in a place where no one will bother them before an hour in.
> 
> And once an hour in hits, all of those can't beat a gygax or a durand.

The only downside is the double-edged sword of being in fort knox and that you can't kill a single, specific item instantly (combat mechs)? lol

**Comment by 4dplanner** [comment]
Needs new review, I changed it up a lot

**Comment by 4dplanner** [comment]
Changelog grows longer

**Comment by ExcessiveUseOfCobblestone** [comment]
die die die die die

**Comment by optimumtact** [comment]
u fricken crab, remove those code: ones, that's what the git logs are for

**Comment by 4dplanner** [comment]
Going to need a testmerge, messing with pulling and stuff

File: code/modules/research/xenobiology/crossbreeding/_misc.dm [hunk 2]
```diff
@@ -4,27 +4,93 @@ Slimecrossing Items
...
+		var/atom/movable/AM = parent
+		var/turf/T = locate(x,y,z)
+		AM.forceMove(T)
+	if(istype(parent, /mob/living/simple_animal))
+		var/mob/living/simple_animal/M = parent
+		M.adjustHealth(health)
+	else if(istype(parent, /obj))
+		var/obj/O = parent
+		O.obj_integrity = health
+	rewinds_remaining -= 1
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/xenobiology/crossbreeding/_misc.dm]:
```suggestion
	rewinds_remaining --
```

File: code/modules/research/xenobiology/crossbreeding/_misc.dm [hunk 3]
```diff
@@ -4,27 +4,93 @@ Slimecrossing Items
...
+		to_chat(parent, "<span class=notice>But the memory falls out of your reach.</span>")
+
+
+
+/obj/item/camera/rewind/afterattack(atom/target, mob/user, flag)
+	if(!on || !pictures_left || !isturf(target.loc))
+		return
+	if(!used)//selfie time
+		if(user == target)
+			to_chat(user, "<span class=notice>You take a selfie!</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/xenobiology/crossbreeding/_misc.dm]: cringed

File: code/modules/research/xenobiology/crossbreeding/_misc.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/xenobiology/crossbreeding/_misc.dm]: @4dplanner

Comment (4dplanner) [reply code/modules/research/xenobiology/crossbreeding/_misc.dm]: oops

File: code/modules/research/xenobiology/crossbreeding/_misc.dm (hunk 3 above)
Comment (optimumtact) [reply code/modules/research/xenobiology/crossbreeding/_misc.dm]: selfie time

**Comment by Angustmeta** [comment]
Tested these today and they seem fine. 

But timefreeze makes you drop items. Can't remember if it was always like that.

**Comment by 4dplanner** [comment]
> Tested these today and they seem fine.
> 
> But timefreeze makes you drop items. Can't remember if it was always like that.

It's always had the item drop aspect, yes. Thanks for the testing!

**Pull request merged**